### ISSUE TITLE
Support replay frame advance in both directions

### ISSUE THREAD

**Issue by Endaris** [panel-attack/panel-game#303]
Currently when pressing A in a paused replay, the replay will advance forward by 1 frame.

Instead `math.sign(playbackSpeed) * 1` should be added to the clock so that if playback speed is negative (and a rollback copy is available), you can rewind frame by frame.
Depends on fixing panel-attack/panel-game#302 first.

**Comment by z26** [comment]
honestly now that 821 is marked as fixed i might try doing this, it sounds useful to me. (came across this issue while searching for the replay controls... would be nice if they were written down somewhere easy to notice) 
right now if you press the frame advance button while rewinding theres weird behavior where forward and backwards movement in time cancel themselves out on odd presses while on even presses the simulation goes forwards one step)

**Comment by z26** [comment]
The way the playback code was written is a bit strange. 

If the playback speed is set to 0 then you press the "go to the next frame" button, after unpausing playback will occur at normal speed despite playbackspeed = 0.

Also, there's a trick to get the playback to occur at 0.5x speed. This is related to the odd behavior I described in my former post. Basically, use the trick above to ensure that playbackspeed = -1, but with maxruns = 1.

**Comment by z26** [comment]
First of all, I don't really know what the post etiquette is like, so sorry about the tripleposting if that's a problem.

The cause for the half speed playback relates to this condition:

    if playbackSpeed == -1 and not GAME.gameIsPaused then
      if P1 and P1.clock > 0 then  and P1.prev_states[P1.clock-1]
        P1:rollbackToFrame(P1.clock-1)
        P1.lastRollbackFrame = -1 -- We don't want to count this as a "rollback" because we don't want to catchup
      end
      
For whatever reason, if max_runs_per_frame is positive while playbackSpeed is negative, then half of the time the required savestate to rewind to can't be found, so the condition prevents rolling back which would crash the game.

Simply using math.sign(playbackSpeed) * 1 like endaris said prevents the state that triggers the issue. It creates other bugs however, so I made a few changes to ensure that Max_runs == playbackspeed at all times, except during manual frame advance.

After that I changed the speeds to -1,0,1,2,3,4,8,16 like endaris suggested on discord. I didn't add 0.5x for now cause I'm indecisive on what approach to take... I think I'd be able to rewrite the frame code so that any positive float value is supported, but I could also only hardcode 0.5x to work (simpler but not as clean)
